core/src/main/java/com/sequenceiq/cloudbreak/service/StackCommonService.java

```diff
@@ -284,6 +313,15 @@ public FlowIdentifier retryInWorkspace(NameOrCrn nameOrCrn, Long workspaceId) {
         return retry.getFlowIdentifier();
     }
 
+    public void cancelInWorkspace(NameOrCrn nameOrCrn, Long workspaceId) {
+        Long stackId = stackService.getIdByNameOrCrnInWorkspace(nameOrCrn, workspaceId);
+
+        LOGGER.info("Cancelling existing flows for the stack : {}", stackId);
+        flowCancelService.cancelRunningFlows(stackId);
```

Comment:
so someone can cancel any flow from any service anytime with an API call? are we sure it is well tested with every operation (uggrade, repair, etc.). we used this internally until this point, we should be exposed it carefully with limitations to avoid various untested scenarios

Reply:
and this is not behind the introduced entitlement as I have seen

Reply:
what is the scenario that an external flow cancel is needed?

Reply:
The cancel api is internal only, so customer can not call it. That is why it is not behind entitlement. We use it for test purposes using this approach https://cloudera.atlassian.net/wiki/spaces/ENG/pages/2082013912/Internal+actor+usage+with+CB+services+using+CBD+Intellij+IDEA

Reply:
yes, i have seen, unfortunately if it is internal, it is still can cause issue, since swagger shows this API and any developer from a CDP service can call it from internal API without any aware of what can be the result of the cancel call, so we need to be extra cautious regarding this logic

Reply:
Looks like I replied further up. We can remove this right now if required (didn't realize in the comment above that it is already behind an entitlement), and figure out a better way to achieve the same.

Reply:
Fixed in the next push (Removed API for now). Resolving conversation.